src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -698,5 +698,66 @@ internal static void UpdateLayerBorder(this CoreAnimation.CALayer layer, IButton
 			if (stroke.CornerRadius >= 0)
 				layer.CornerRadius = stroke.CornerRadius;
 		}
+
+		internal static T? FindResponder<T>(this UIView view) where T : UIResponder
+		{
+			var nextResponder = view as UIResponder;
+			while (nextResponder is not null)
+			{
+				nextResponder = nextResponder.NextResponder;
+
+				if (nextResponder is T responder)
+					return responder;
+			}
+			return null;
+		}
+
+		internal static UIView? FindNextView(this UIView view, UIView superView, Func<UIView, bool> isValidType)
+		{
+			var passedOriginal = false;
+
+			var nextView = superView.FindNextView(view, ref passedOriginal, isValidType);
```

Comment:
Instead of always starting at the top level view and searching down to find your place can we start the search from where we are at and search out? This way we aren't always traversing the entire view stack?